### DISCUSSION TITLE
关于 Proxy Pool 满载问题及部署建议的公告 / Announcement Regarding Proxy Pool Overload and Deployment Suggestions

### DISCUSSION THREAD

**Discussion by vvbbnn00** [vvbbnn00/WARP-Clash-API#112] [Announcements]
首先，非常感谢大家对本项目的认可和支持！
近日以来，随着用户数量的增加，我们注意到项目部署的IP代理池访问量激增，服务器已经连续数日处于满载状态。这可能会导致用户部署的实例无法正常刷取WARP流量。
为了解决这一问题，我们计划在近日对 proxy_pool 获取节点相关的接口进行性能优化。在优化完成之前，为了保证服务的正常运行，我们将开启Cloudflare的缓存功能。请注意，这可能会对获取到的节点IP的随机性产生较大影响。
因此，我们强烈建议有能力的用户自行部署 proxy_pool 实例，并通过修改 PROXY_POOL_URL 环境变量，将IP代理池指向自行搭建的服务，以减轻服务器的负担。

感谢您的支持和理解！

----

First and foremost, we would like to express our sincere gratitude for your recognition and support of our project! 
Recently, with the increase in the number of users, we have noticed a significant surge in the traffic to the IP proxy pool deployed by us. The server has been overloaded for several consecutive days, which may cause issues in refreshing WARP traffic for user-deployed instances.
To address this issue, we plan to optimize the performance of the interfaces related to node retrieval in the proxy_pool in the coming days. Before the optimization is completed, to ensure the normal operation of the service, we will enable the caching function of Cloudflare. Please note that this may have a significant impact on the randomness of the obtained node IPs.
Therefore, we strongly recommend users who are capable to deploy their own proxy_pool instances and modify the PROXY_POOL_URL environment variable to point the IP proxy pool to their own service, in order to alleviate the burden on the server. 

Thank you for your support and understanding!

**Comment by vvbbnn00** [comment]
我们已经使用Go语言重构了API接口并上线。
由于当前需求相对简单，目前我们只实现了获取代理IP的接口。值得一提的是，经过重构后，CPU的占用有了显著的下降，具体表现如下图所示。
相关实现代码已开源，可供参考：https://github.com/vvbbnn00/proxy_pool

----

We have already restructured the API interface using Go language and launched it online. 
As the current demand is relatively simple, we have only implemented the interface for obtaining proxy IPs for now. It is worth mentioning that after restructuring, the CPU usage has significantly decreased, as shown in the following figure. 
The relevant implementation code has been open-sourced and is available for reference: https://github.com/vvbbnn00/proxy_pool


![image](https://github.com/vvbbnn00/WARP-Clash-API/assets/46409975/f8293b60-c1b0-4457-a675-dd5cbdfbb8f6)

**Reply by Carlos920** [reply]
![image](https://github.com/vvbbnn00/WARP-Clash-API/assets/32257015/6bf2413a-211b-42ab-b31e-5fdea7f8577d)
阿里云上构建失败，go_api的docker文件里我添加了国内的源，但是还是不行

**Reply by vvbbnn00** [reply]
之前仓库里漏了`go.sum`，已加上，重新pull以下应该可以了

**Reply by Carlos920** [reply]
有个问题，我部署好proxy pool 后 在`.env.local`文件中也添加了自己的地址，但是刷新流量任然报429错误，这个是需要重置账号吗？
![image](https://github.com/vvbbnn00/WARP-Clash-API/assets/32257015/985401ab-4927-4ae2-925f-781822e7939b)

**Reply by vvbbnn00** [reply]
如果您的账号已经刷满流量了（即24TB），那么将不能再获取新流量，返回429；
还有一种可能是您的proxy_pool项目获取到的IP和我部署的项目获取的IP相同（因为都是从免费池获取的），而这个IP在短期内已经达到了上限，这种情况的话，您可以尝试更换一些其他的代理IP源（需要修改项目代码），或者，您也可以尝试直接使用刷好流量的`licensekey`

**Comment by home2020** [comment]
部署完了还会占用这个吗？

**Comment by Carlos920** [comment]
> 之前仓库里漏了`go.sum`，已加上，重新pull以下应该可以了

解决了，thanks

**Comment by dezhishen** [comment]
咨询下
`docker-compose` 里面的 `flaresolverr` 如何使用已经部署的。

**Reply by vvbbnn00** [reply]
如果是不同`docker-compose`项目之间的通信，可以尝试建立一个`docker network`，然后在下方代码的位置替换您的`url`

https://github.com/vvbbnn00/proxy_pool/blob/e1eb07abba9001ad790e81a03ca144f376139cd0/util/proxyBypassRequest.py#L30

**Reply by dezhishen** [reply]
能否提供环境变量？类似redis

**Reply by vvbbnn00** [reply]
等早上我改一下吧
